compiler/testData/debug/stepping/inlineCallableReference.kt

```diff
@@ -13,8 +13,11 @@ inline fun f(block: () -> Unit) {
 // LINENUMBERS
 // test.kt:3
 // test.kt:4
-// test.kt:10
+// test.kt:10 @ test.kt:4
 // test.kt:5
 // test.kt:6
-// test.kt:11
+// test.kt:11 @ test.kt:4
 // test.kt:7
+
+// line 11 has incorrect call site (:5)
```

Comment:
@max-kammerer small problem here. In this code, for example
```
fun main() {
    f()
}

inline fun f() {
    println("...")
    throw RuntimeError("...")
}
```
the SMAP contains the following entry in the KotlinDebug stratum:
```
2#1,2:9
```
The problem is that according to JSR-045, this actually means "map line 9 to 2 and line 10 to 3", which is why these tests fail. However, if you paste this code into IDEA and click "navigate to inline function call site", it will map line 10 to 2, apparently interpreting this line to mean "map *both* 9 and 10 to 2" because it's in the call site stratum.

Unfortunately, it seems like the only way to emit JSR-045-correct data here is to map each line separately:
```
2#1:9
2#1:10
```
because something like
```
2#1,2:9,0
```
actually means "map line 9 to range [2,2) and line 10 to range [2,2)", i.e. it actually maps each line to precisely nothing.

Obviously, this will increase the size of SMAPs when inlining large functions. However, I assume that this increase in size will not be noticeable in jars because "2#1:" is repeated and thus deflate should be able to compress it. What do you think?

Reply:
JDI doesn't know what `2#1:9,2` means either.

So much for following specifications...

Reply:
> which is the correct way to say "map lines 9 and 10 to 2" is not supported by IDEA 

Maybe it's lack in Kotlin Debugger support and could be simple fixed // cc  @yandex

Reply:
You mean @yanex? :)

Reply:
> JDI doesn't know what 2#1:9,2 means either and simply ignores the ,2 part.

it's quite strange, it's definetly should be a issue. Maybe it's fixed in later versions of JDK

Reply:
It does support this syntax after all. The problem is that so far I don't see an API to correctly retrieve the line number from KotlinDebug:
1. like the comment in this PR says, if an unmapped line number does not appear in the stratum, JDI will return "undefined garbage", or more specifically it will attempt to guess the closest line number by bytecode position instead of just saying that there's no line number: https://github.com/AdoptOpenJDK/openjdk-jdk/blob/master/src/jdk.jdi/share/classes/com/sun/tools/jdi/LocationImpl.java#L159
2. the `allLineLocations` hack I'm using to work around this intentionally deduplicates the line numbers: https://github.com/AdoptOpenJDK/openjdk-jdk/blob/master/src/jdk.jdi/share/classes/com/sun/tools/jdi/ConcreteMethodImpl.java#L308 but the equality comparison for LineStratum is "same SMAP stratum, same mapping, and maps to the same result" - so lines 9 and 10 that are both remapped by the same `2#1:9,2` to line 2 are considered equal, thus only 9 is returned, and the test no longer tries to print the call site for line 10.